autonat.go

```diff
@@ -63,21 +47,30 @@ type autoNATResult struct {
 	address ma.Multiaddr
 }
 
-// NewAutoNAT creates a new ambient NAT autodiscovery instance attached to a host
-// If getAddrs is nil, h.Addrs will be used
-func NewAutoNAT(ctx context.Context, h host.Host, getAddrs GetAddrs) AutoNAT {
-	if getAddrs == nil {
-		getAddrs = h.Addrs
+// New creates a new NAT autodiscovery system attached to a host
+func New(ctx context.Context, h host.Host, options ...Option) (AutoNAT, error) {
+	conf := new(config)
+
+	if err := defaults(conf); err != nil {
+		return nil, err
+	}
+	if conf.getAddressFunc == nil {
+		conf.getAddressFunc = h.Addrs
 	}
```

Comment:
I'd either have this come after the options or have the `WithAddresses` option return an error if `nil` is passed in. The latter is probably better.